lib/fast-path.js

```diff
@@ -220,6 +220,11 @@ FPp.needsParens = function(assumeExpressionContext) {
     return false;
   }
 
+  if (parent.type === "ArrowFunctionExpression" && name === "body") {
+    // TODO does this expression start with a {?
```

Comment:
It can get way more complicated than that.  For example: https://astexplorer.net/#/gist/a679dafedca515172665db0851858392/39372cf4bf7df510adca43b5bbdb9abd9e9dd1c1

In the example, `node.type === 'BinaryExpression'` but `node.left.left.left.object.type === 'ObjectExpression'`.

Reply:
We could do something like `if(printGenerically(node)).match(/\s*\{/))`.  It's inefficient but pretty simple.